### PR TITLE
Luwei/onednn3.2 migration

### PR DESCRIPTION
**Pull request by luweizhou2016**
-Merge all the forked patched in 3.1.
-Fix zero point issue.
-Fix prelu post ops bug.
-Revert jit-recorder tailing optimization patch because windows compiler issue.
 https://jira.devtools.intel.com/browse/MFDNN-10233
-Revert onednn smallic falling back to jit from brgemm to fix perf 5% regression on ICX.
  https://jira.devtools.intel.com/browse/MFDNN-8748
### ONEDNN branch
https://github.com/openvinotoolkit/oneDNN/tree/luwei/v3.2_for_ie_master_migration_squash
https://github.com/openvinotoolkit/oneDNN/commits/luwei/v3.2_for_ie_master_migration_squash
### Tickets:
 - *99138, 11223, 110341*

### CODE REVIEW

File: src/frontends/onnx/tests/onnx_import_com_microsoft.in.cpp [hunk 1]
```diff
@@ -27,36 +27,6 @@ using namespace ngraph;
 static std::string s_manifest = "${MANIFEST}";
 static std::string s_device = test::backend_name_to_device("${BACKEND_NAME}");
 
-NGRAPH_TEST(${BACKEND_NAME}, onnx_model_bias_gelu) {
```
Comment (EgorDuplensky) [inline src/frontends/onnx/tests/onnx_import_com_microsoft.in.cpp:30]: Why are we deleting this test?

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/softmax.cpp [hunk 2]
```diff
@@ -190,16 +190,11 @@ const std::vector<SoftMaxConfig> unsupportedConfigsFP32{
...
-const auto avx512 = CPUSpecificParams{{}, {}, {"jit_avx512"}, "jit_avx512"};
-const auto avx2 = CPUSpecificParams{{}, {}, {"jit_avx2"}, "jit_avx2"};
-const auto sse42 = CPUSpecificParams{{}, {}, {"jit_sse42"}, "jit_sse42"};
-const auto ref = CPUSpecificParams{{}, {}, {"ref_any"}, "ref_any"};
 
-const std::vector<CPUSpecificParams> vecCpuConfigs = {ref, sse42, avx2, avx512};
 const auto OptimizedParams = testing::Combine(testing::Values(ElementType::f32, ElementType::bf16),
                                               testing::ValuesIn(optimizedConfigsFP32),
                                               testing::Values(CommonTestUtils::DEVICE_CPU),
-                                              testing::ValuesIn(filterCPUInfoForDevice(vecCpuConfigs)));
```
Comment (EgorDuplensky) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/softmax.cpp:202]: What is the motivation here?

File: src/frontends/onnx/tests/onnx_import_com_microsoft.in.cpp (hunk 1 above)
Comment (luweizhou2016) [reply src/frontends/onnx/tests/onnx_import_com_microsoft.in.cpp:30]: The error is a little higher than default threshold. I didn't have time to find out how to set the threshold. so comment for now.

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/convolution.cpp [hunk 3]
```diff
@@ -1648,67 +1648,6 @@ INSTANTIATE_TEST_SUITE_P(smoke_Conv_2D_AutoPad_FP32, ConvolutionLayerCPUTest,
 
 } // namespace
 
-#if !defined(__clang__) || defined(__APPLE__)
```
[the remark this answers is not in the document]
Comment (luweizhou2016) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/convolution.cpp:1651]: Yes. Winograd conv is removed

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/softmax.cpp (hunk 2 above)
Comment (luweizhou2016) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/softmax.cpp:202]: 3.1 onednn softmax list:
            CPU_INSTANCE_X64(jit_uni_softmax_fwd_t<avx512_core>)
            CPU_INSTANCE_X64(jit_uni_softmax_fwd_t<avx2>)
            CPU_INSTANCE_X64(jit_uni_softmax_fwd_t<sse41>)  
3.2 softmax list:
CPU_INSTANCE_X64(jit_uni_softmax_fwd_t)

unified jit_uni_softmax_fwd_t.

**Comment by luweizhou2016** [comment]
> Could you please add the link to oneDNN fork branch to the PR description details.

https://github.com/openvinotoolkit/oneDNN/tree/luwei/v3.2_for_ie_master_migration_squash

**Review by EgorDuplensky — COMMENTED** [review]
@luweizhou2016
It seems, that in order to fix your current build issue you need to replace
DNNL_USE_ACL
with
DNNL_AARCH64_USE_ACL
in thirdparty/CMakeLists.txt file

**Comment by luweizhou2016** [comment]
@EgorDuplensky , Pls help to review. I am thinking to merge this PR next week if no big perf issue. Thx

**Review by EgorDuplensky — COMMENTED** [review]
the code changes lgtm.
I have not found the results of the regression checks in the mentioned tickets though.
We need to add the regression check results for X86 and ARM at one of the tickets.
112237 this one for example.

**Comment by maxnick** [comment]
@mateusztabaka , are you okay with the change in the onnx frontend test?

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/softmax.cpp (hunk 2 above)
Comment (maxnick) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/softmax.cpp:202]: These tests were aimed toward testing our custom modifications to the softmax primitive. Are they removed in this oneDNN version?

File: src/plugins/intel_cpu/src/nodes/deconv.cpp [hunk 4]
```diff
@@ -1012,8 +1010,7 @@ void Deconvolution::createDescriptor(const std::vector<MemoryDescPtr> &inputDesc
     } else {
         dnnl::memory::desc wgh_candidate(DnnlExtensionUtils::convertToDnnlDims(getWeightDims()),
                                            dnnlInDesc.getDataType(), memory::format_tag::any);
-        for (auto alg : {dnnl::algorithm::convolution_winograd, dnnl::algorithm::convolution_direct}) {
-        // for (auto alg : {dnnl::algorithm::convolution_direct}) {
+        for (auto alg : {dnnl::algorithm::convolution_direct}) {
```
[the remark this answers is not in the document]
Comment (luweizhou2016) [reply src/plugins/intel_cpu/src/nodes/deconv.cpp]: Applied

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/softmax.cpp (hunk 2 above)
Comment (luweizhou2016) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/softmax.cpp:202]: File https://jira.devtools.intel.com/browse/CVS-114609 to track on this.

**Comment by maxnick** [comment]
@EgorDuplensky , do you have any further comments?

**Pull request merged**